### PR TITLE
clblast: init at 1.5.3

### PR DESCRIPTION
**Pull request by joachimschmidt557**
###### Description of changes

Add [CLBlast](https://github.com/CNugteren/CLBlast/tree/master), a library providing a BLAS API with an OpenCL backend. Used in e.g. [ggml](https://github.com/ggerganov/ggml).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/science/math/clblast/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "clblast";
+  version = "1.5.3";
+
+  src = fetchFromGitHub {
+    owner = "CNugteren";
+    repo = "CLBlast";
+    rev = "${version}";
```
Comment (natsukium) [inline pkgs/development/libraries/science/math/clblast/default.nix:17]:
```suggestion
    rev = version;
```
or `rev = "refs/tags/${version}";`

File: pkgs/development/libraries/science/math/clblast/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    sha256 = "0a4rfpg5523ckcvfarkpqaq6crhg9nnkyr2bj0j07w3dapx6slyl";
```
Comment (natsukium) [inline pkgs/development/libraries/science/math/clblast/default.nix:18]:
```suggestion
    hash = "0a4rfpg5523ckcvfarkpqaq6crhg9nnkyr2bj0j07w3dapx6slyl";
```

File: pkgs/development/libraries/science/math/clblast/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  buildInputs = lib.optionals (!stdenv.isDarwin) [
+    opencl-headers
+    ocl-icd
+  ] ++ lib.optionals stdenv.isDarwin [
+    OpenCL
+  ];
+
+  meta = with lib; {
+    homepage = "https://cnugteren.github.io/clblast";
+    description = "The tuned OpenCL BLAS library";
```
Comment (natsukium) [inline pkgs/development/libraries/science/math/clblast/default.nix:38]: Would you please add `meta.changelog`?

File: pkgs/development/libraries/science/math/clblast/default.nix (hunk 2 above)
Comment (joachimschmidt557) [reply pkgs/development/libraries/science/math/clblast/default.nix]: I changed it to an SRI hash.

File: pkgs/development/libraries/science/math/clblast/default.nix (hunk 3 above)
Comment (joachimschmidt557) [reply pkgs/development/libraries/science/math/clblast/default.nix:39]: Added

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 232897` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>clblast</li>
  </ul>
</details>

File: pkgs/development/libraries/science/math/clblast/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  postPatch = ''
+    substituteInPlace clblast.pc.in \
+      --replace "\''${prefix}/" "" \
+      --replace "\''${exec_prefix}/" "" \
+  '';
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = lib.optionals (!stdenv.isDarwin) [
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/math/clblast/default.nix:29]:
```suggestion
  buildInputs = lib.optionals stdenv.isLinux [
```

File: pkgs/development/libraries/science/math/clblast/default.nix [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    owner = "CNugteren";
+    repo = "CLBlast";
+    rev = version;
+    hash = "sha256-1FNt+lVt8AMkkEtkP61ND2ZmsMJ3ZuU2m2yIUt51mSg=";
+  };
+
+  postPatch = ''
+    substituteInPlace clblast.pc.in \
+      --replace "\''${prefix}/" "" \
+      --replace "\''${exec_prefix}/" "" \
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/math/clblast/default.nix:24]: This does not look like it will work

Comment (joachimschmidt557) [reply pkgs/development/libraries/science/math/clblast/default.nix:24]: It does:

```
joachim@dustbowl ~/s/n/result> pwd
/home/joachim/src/nixpkgs-clblast/result
joachim@dustbowl ~/s/n/result> cat lib/pkgconfig/clblast.pc
prefix=/nix/store/c8g193qg06crwmzlnmsd8xnfmyx88bjy-clblast-1.5.3
exec_prefix=${prefix}
includedir=/nix/store/c8g193qg06crwmzlnmsd8xnfmyx88bjy-clblast-1.5.3/include
libdir=/nix/store/c8g193qg06crwmzlnmsd8xnfmyx88bjy-clblast-1.5.3/lib

Name: CLBlast
Description: CLBlast is a modern, lightweight, performant and tunable OpenCL BLAS library written in C++11
Version: 1.5.3
Libs: -L${libdir} -lclblast
Cflags: -I${includedir}
joachim@dustbowl ~/s/n/result> 
```

`exec_prefix` hast been removed from `libdir`.

File: pkgs/development/libraries/science/math/clblast/default.nix (hunk 4 above)
Comment (joachimschmidt557) [reply pkgs/development/libraries/science/math/clblast/default.nix]: Changed :+1:

File: pkgs/development/libraries/science/math/clblast/default.nix (hunk 5 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/science/math/clblast/default.nix:24]: I am not sure if this is the best solution here. A patch would make this clearer.

**Review by SuperSandro2000 — COMMENTED** [review]
please rebase

**Pull request closed without merging**